pymc3/distributions/continuous.py

```diff
@@ -1654,6 +1655,84 @@ def logcdf(self, value):
         )
 
 
+class AsymmetricLaplace(Continuous):
+    r"""
+    Asymmetric-Laplace log-likelihood.
+
+    The pdf of this distribution is
+
+    ..math::
+        {f(x|\\b,\kappa) =
+            \left({\frac{\\b}{\kappa + 1/\kappa}}\right)\,e^{-(x)\\b\,s\kappa ^{s}}}
+        where s = sgn(x)
+
+
+    See also: https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution
+    """
+
+    def __init__(self, b, kappa, testval=0.0, *args, **kwargs):
+        self.b = tt.as_tensor_variable(b)
+        self.kappa = tt.as_tensor_variable(kappa)
+
+        super().__init__(*args, **kwargs, testval=testval)
+
+    def _random(self, b, kappa, size=None):
+        if size is not None:
+            u = np.random.uniform(size=size)
+            x = -np.log((1 - u) * (1 + kappa ** 2)) / (kappa * b) * (
+                u > ((kappa ** 2) / (1 + kappa ** 2))
+            ) + kappa * np.log(u * (1 + kappa ** 2) / (kappa ** 2)) / b * (
+                u < ((kappa ** 2) / (1 + kappa ** 2))
+            )
+            return x
+
+        u = np.random.uniform()
+        if u > (kappa ** 2) / (1 + kappa ** 2):
+            x = -np.log((1 - u) * (1 + kappa ** 2)) / (kappa * b)
+        else:
+            x = kappa * np.log(u * (1 + kappa ** 2) / (kappa ** 2)) / b
+
+        return x
+
+    def random(self, point=None, size=None):
+        """
+        Draw random samples from this distribution, using the inverse CDF method.
+
+        Parameters
+        ----------
+        point: dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size:int, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+        """
+        b, kappa = draw_values([self.b, self.kappa], point=point, size=size)
+        return generate_samples(self._random, b, kappa, dist_shape=self.shape, size=size)
+
+    def logp(self, value):
+        """
+        Calculate log-probability of Asymmetric-Laplace distribution at specified value.
+
+        Parameters
+        ----------
+        value: numeric
+            Value(s) for which log-probability is calculated. If the log probabilities for multiple
+            values are desired the values must be provided in a numpy array or theano tensor
+
+        Returns
+        -------
+        TensorVariable
+        """
+        return tt.log(self.b / (self.kappa + (self.kappa ** -1))) + (
```

Comment:
Is this function dealing properly with invalid parameters, such as a negative `b` and `kappa`? If not you might need to wrap the result in a `bound`

Reply:
I tried running this function on some  invalid parameters and It does not return any error, instead it prints the corresponding values. I guess, it can handle the invalid parameters. 
If in some sense this is wrong, please do let me know.
I am, hoping that I got your question correctly.

Reply:
Does it return `-inf` for invalid parameters?

Reply:
for negative `kappa`, it returns `nan`, when `kappa` is 0, it returns `-inf`. same goes for `b`

Reply:
Then I think you should indeed wrap the results in a bound. Something like this:

```python
return bound(
    <your logp code here>,
    0 < b,
    0 < kappa,
)
```

This will replace the `nan` by `-inf` when using invalid negative parameters